pkg/sql/sem/builtins/geo_builtins.go

```diff
@@ -2285,6 +2285,51 @@ Note if geometries are the same, it will return the LineString with the minimum
 		},
 	),
 
+	//
+	// Geography operations
+	//
+	"st_geohash": makeBuiltin(
+		defProps(),
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"geography", types.Geography},
```

Comment:
PostGIS seems to enable this to work on both Geometry and Geography types. mind doing that here? Also let's you resolve https://github.com/cockroachdb/cockroach/issues/48943!
Since it works for both geometry and geography, you may want to move the function from geogfn to just geo.